src/interaction/TrackManager.js

```diff
@@ -530,6 +530,20 @@ class TrackManager {
     this.notifyTrackChangeEventListeners_();
   }
 
+  /**
+   * Delete a POI and notify track change listeners.
+   * @param {number} index
+   */
+  deletePOI(index) {
+    const poi = this.trackData_.getPOIs().find(feature => feature.get('index') === index);
+    const feature = this.source_.getFeatures().find(feature => feature.get('index') === index);
+    this.source_.removeFeature(feature);
+    this.trackData_.deletePOI(poi);
+    this.trackData_.updatePOIIndexes();
+    this.submode = '';
```

Comment:
@RenataMuellerC2C, why do you change the submode?
I guess this is app-dependent and should be done by the calling code.